config/initializers/github_client.rb

```diff
@@ -2,12 +2,22 @@
 
 module GitHubClassroom
   def self.github_client(options = {})
-    client_options = {
-      client_id:     Rails.application.secrets.github_client_id,
-      client_secret: Rails.application.secrets.github_client_secret,
-      auto_paginate: true
-    }.merge!(options)
+    client_options = options.tap do |opts|
+      opts[:user_agent] = "GitHub Classroom"
 
+      unless options.key?(:access_token)
+        opts[:client_id]     = Rails.application.secrets.github_client_id
+        opts[:client_secret] = Rails.application.secrets.github_client_secret
+      end
+
+      if enterprise_instance?
+        opts[:api_endpoint] = "https://#{Rails.application.secrets.github_enterprise_hostname}/api/v3"
+      end
+    end
     Octokit::Client.new(client_options)
   end
+
+  def self.enterprise_instance?
```

Comment:
Is there any reason we couldn't just use a feature flag to enable enterprise support instead of adding another secret? That would follow the pattern of our other new features and would make it easier to toggle this

Reply:
Definitely! Could you explain this a bit better? Currently, I've added two environment variables (See `.env.example` / `ENTERPRISE_ENABLED`) and just following what I saw other code doing at the time, it looked like the way to get environment variables down to Rails was through secrets. 

If there's a way to query `ENTERPRISE_ENABLED` inside of this method, I'm 100% down to get rid of the secrets and just read from `ENV`.

Reply:
I was thinking we could add a feature flag in https://github.com/mdwheele/classroom/blob/feature/support-ghe/app/controllers/application_controller/feature_flags_dependency.rb to check if we want to run on an enterprise instance.

Reply:
I've refactored this a bit... now the feature is enabled based on whether or not the hostname is present in the environment. In order for the feature to work we need that hostname in the env and because of this I think it makes sense to hold off on introducing a feature flag.